pkg/collectors/builder.go

```diff
@@ -297,6 +311,24 @@ func (b *Builder) startWatchingManifestWorks() {
 		klog.Fatalf("cannot create workclient: %v", err)
 	}
 
+	// refresh the manifestwork store once the cluster ID of a certian cluster is changed
+	b.clusterIdCache.AddOnClusterIdChangeFunc(func(clusterName string) error {
+		klog.Infof("Refresh the manifestwork metrics since the cluster ID of cluster %q is changed", clusterName)
+		works, err := workClient.WorkV1().ManifestWorks(clusterName).List(b.ctx, metav1.ListOptions{})
```

Comment:
do we need to call client?

Reply:
We have to call the client to list the objects in a certain namespace unless we want to cache all objects in memory. Caching all objects, e.g. ManifestWorks, will consume a lot of memory. This function will only be called when
1) a new cluster is added into the cache;
2) the cluster id label is added to a managedcluter;

It will not happen frequently. So I think it will not result in too much pressure on the kube-apiserver.